### DISCUSSION TITLE
Automatically detect possible block positions

### DISCUSSION THREAD

**Discussion by MekDrop** [ImpressCMS/impresscms#1044] [Ideas]
Like WordPress does. There is no need to have the admin area for create new positions - there is no point to have more or less than theme was designed to support.

Idk... how to implement that?

**Comment by MekDrop** [comment]
Maybe we could add optional field in composer.json theme config `blocksPositions`. If it defined, it will list all supported block positions f.e.:
```
{
 ...
  "extras": {
   ...
   "blocksPositions": [
     "left",
     "right",
     "banner"
   ]
  }
}
```

If not defined ICMS will think that this theme supports all default block positions, so for non composer themes allways will be accessible only default block positions.

If there was a previsouly assigned theme that has some block position that doesn't exist anymore, all these blocks will be automatically assigned for the first block position that exists in current theme.

And than we could remove Block positions configuration from admin.

Do you think that would be ok?

**Reply by davidimpress** [reply]
So - there would be a default setting with the "legacy" positions? And the config could add the additional?
Sounds interesting. How tricky to edit this, or is it not required?
How difficult to assign a block to a different location?

**Reply by fiammybe** [reply]
Another good element in ImpressCMS that is widely under-used : the block positions :-) I think the block positions should be stored based on the theme. Once you go out of the standard positions, you have no certainty that the names used in the theme will be the same, so if you change from one theme to another, you don't want to start reconfigure your site manually. Let's say you want to change temporarily from the standard theme to a christmas theme: being able to pre-configure the block positions for that theme would be a huge plus.
And having a 'default' block will be a requirement I think. 

On the other hand, do we need to configure that in the composer file? Can't we parse the theme.html and just detect the block positions?
We stuff lots of things in the composer file, it will make people afraid of creating themes or addons if it becomes too complex
